data/scripts.js

```diff
@@ -1353,7 +1358,7 @@ exports.BattleScripts = {
 				}
 
 				// Remove rejected moves from the move list.
-				if (rejected && movePool.length) {
+				if (rejected && movePool.length - availableHP + ((!availableHP || hasMove['hiddenpower'] && move.id !== 'hiddenpower') ? 0 : 1) >= 5 - moves.length) {
```

Comment:
Thing is, the rejection code in that line is always set up to only remove one move at a time, because the `break` command in there causes that inner loop to quit and the outer loop to re-add moves starts again. So checking `5 - moves.length` is wholly unneccessary.

And since you only ever need to check for 1 slot, when you remove Hidden Power and there are available HP's left, `movePool.length - availableHP + 1 >= 1` always has the same results as just checking `movePool.length >= 1`.